File: japan-go-fe/types/api/responses/lesson.response.ts
import { BaseResponse } from "./base.response";
import { GrammarResponse } from "./grammar.response";
import { KanjiPageResponse } from "./kanji.response";
import { LessonType } from "@/types/enums/lesson.enum";

export interface LessonResponse extends BaseResponse<number> {
    lessonName: string;
    lessonType: LessonType;
    grammarLesson: GrammarLessonResponse;
    kanjiLesson: KanjiLessonResponse;
}

export interface GrammarLessonResponse extends BaseResponse<number> {
    grammarLessonTitle: string;
    grammars: GrammarResponse[];
}

export interface KanjiLessonResponse extends BaseResponse<number> {
    kanjiPages: KanjiPageResponse[];
}

File: japan-go-be/core/service/src/main/java/org/japan/dto/response/lesson/LessonResponse.java
package org.japan.dto.response.lesson;

import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;
import org.japan.constants.lesson.LessonTypeEnum;
import org.japan.dto.response.base.BaseResponse;

@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class LessonResponse extends BaseResponse {
    String lessonName;
    String description;
    LessonTypeEnum lessonType;
    Long pageCount;
    GrammarLessonResponse grammarLesson;
    KanjiLessonResponse kanjiLesson;
    BookResponse book;
}
